File: newver/utils/files.py
from pathlib import Path
import shutil

def copy_cue_file(original_bin: Path, patched_bin: Path) -> Path | None:
    """
    Copy the .cue file corresponding to the original BIN and rename it
    to match the patched BIN. Returns the new cue path, or None if no cue exists.
    """
    orig_cue = original_bin.with_suffix(".cue")
    if orig_cue.exists():
        patched_cue = patched_bin.with_suffix(".cue")
        shutil.copy(orig_cue, patched_cue)
        return patched_cue
    return None

File: cpp/src/utils/Files.cpp
#include "Files.h"
#include <QFile>
#include <QFileInfo>

namespace Files {

QString copyCueFile(const QString &originalBin, const QString &patchedBin)
{
    QFileInfo origInfo(originalBin);
    QString origCue = origInfo.path() + "/" + origInfo.completeBaseName() + ".cue";

    if (!QFile::exists(origCue))
        return {};

    QFileInfo patchedInfo(patchedBin);
    QString patchedCue = patchedInfo.path() + "/" + patchedInfo.completeBaseName() + ".cue";

    if (QFile::exists(patchedCue))
        QFile::remove(patchedCue);

    if (QFile::copy(origCue, patchedCue))
        return patchedCue;

    return {};
}

}